### ISSUE TITLE
linter: unused variable if used in a sequence expression

### ISSUE THREAD

**Issue by sam-goodwin** [oxc-project/oxc#15174]
### What version of Oxlint are you using?

0.9.0

### What command did you run?

"[typescript]": {     "editor.indentSize": 2,     "editor.defaultFormatter": "oxc.oxc-vscode"   },

### What does your `.oxlintrc.json` config file look like?

```jsonc
// (none)
```

### What happened?

I have a function that uses a sequence (aka comma) expression but Ox lint thinks it is unused
```ts
items.reduce((acc, item) => (acc[item.action]++, acc)
```

```ts
Parameter 'item' is declared but never used. Unused parameters should start with a '_'.
help: Consider removing this parameter.oxc[eslint(no-unused-vars)](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-vars.html)
```
